src/validation.cpp

```diff
@@ -2194,29 +2207,32 @@ class ConnectTrace {
     }
 
     void BlockConnected(CBlockIndex* pindex, std::shared_ptr<const CBlock> pblock) {
-        blocksConnected.emplace_back(pindex, std::move(pblock));
-        conflictedTxs.emplace_back();
-    }
-
-    std::vector<std::pair<CBlockIndex*, std::shared_ptr<const CBlock> > >& GetBlocksConnected() {
+        assert(!blocksConnected.back().pindex);
+        assert(pindex);
+        assert(pblock);
+        blocksConnected.back().pindex = pindex;
+        blocksConnected.back().pblock = std::move(pblock);
```

Comment:
I think given where we call BlockConnected you could make pblock be an rvalue and move from the caller, saving the pass by copy.